File: PanKanapka/Api.Domain/Models/Reservation.cs
using System.Collections.Generic;

namespace Api.Domain.Models
{
    public class Reservation
    {
        public long ReservationId { get; set; }
        public long ClientFirmId { get; set; }
        public bool IsActiveReservation { get; set; }
        public IEnumerable<ReservationItem> ReservationItems { get; set; }
    }

    public class ReservationItem
    {
        public string Name { get; set; }
        public int Quantity { get; set; }
        public double PriceTotal { get; set; }
        public string ItemImageUrl { get; set; }
    }


    public class ClientReservation
    {
        public long ReservationId { get; set; }
        public CateringFirm cateringFirm { get; set; }
        public bool IsActiveReservation { get; set; }
        public string Name { get; set; }
        public int Quantity { get; set; }
        public double PriceTotal { get; set; }
        public string ItemImageUrl { get; set; }
    }

    public class CateringFirm
    {
    }
}

File: pankan_appka/lib/api/models/reservation.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pankan_appka/api/models/reservation_item.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  final int reservationId;
  final int clientFirmId;
  final bool isActiveReservation;
  final List<ReservationItem> reservationItems;

  Reservation({this.reservationId, this.clientFirmId, this.isActiveReservation, this.reservationItems});

  static fromJson(Map<String, dynamic> jsonMap) => _$ReservationFromJson(jsonMap);

  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}